app/views/course/assessment/question/programming/_question.json.jbuilder

```diff
@@ -1,41 +1,42 @@
 # frozen_string_literal: true
 json.question do
-  json.(@programming_question, :id, :title, :staff_only_comments, :maximum_grade,
-        :language_id, :memory_limit, :time_limit, :is_low_priority)
-  json.max_time_limit @programming_question.max_time_limit
-  json.description format_ckeditor_rich_text(@programming_question.description)
-  json.languages Coursemology::Polyglot::Language.all.order(:name) do |lang|
-    json.(lang, :id, :name)
-    json.editor_mode lang.ace_mode
-  end
-  json.skill_ids @question_assessment.skills.order_by_title.as_json(only: [:id, :title])
-  json.skills current_course.assessment_skills.order_by_title do |skill|
-    json.(skill, :id, :title)
-  end
+  json.partial! 'course/assessment/question/form',
+                question: @programming_question,
+                question_assessment: @question_assessment
 
-  has_submissions = @programming_question.answers.without_attempting_state.count > 0
+  json.skillIds @question_assessment.skills.order_by_title.as_json(only: [:id, :title])
+
+  json.languageId @programming_question.language_id || ''
+  json.memoryLimit @programming_question.memory_limit || ''
+  json.timeLimit @programming_question.time_limit || ''
+  json.maxTimeLimit @programming_question.max_time_limit || ''
+  json.attemptLimit @programming_question.attempt_limit || ''
+  json.isLowPriority @programming_question.is_low_priority
   json.autograded @programming_question.persisted? ? @programming_question.attachment.present? : @assessment.autograded?
-  json.has_auto_gradings @programming_question.auto_gradable? && has_submissions
-  json.has_submissions has_submissions
-  json.display_autograded_toggle display_autograded_toggle?
-  json.course_gamified current_course.gamified?
-  json.autograded_assessment @assessment.autograded?
-  json.published_assessment @assessment.published?
-  json.attempt_limit @programming_question.attempt_limit
-  json.is_codaveri @programming_question.is_codaveri
-  json.codaveri_enabled current_course.component_enabled?(Course::CodaveriComponent)
-  json.existing_submissions_count @assessment.submissions.count
+  json.editOnline can_edit_online?
+
+  has_submissions = @programming_question.answers.without_attempting_state.count > 0
+  json.hasAutoGradings @programming_question.auto_gradable? && has_submissions
+  json.hasSubmissions has_submissions
+
+  autograded_assessment = @assessment.autograded?
+  json.shouldWarnOnSubmit current_course.gamified? && autograded_assessment && has_submissions
```

Comment:
I know this is based on the previous logic. Could this be updated to `has_submission` only? Also on the frontend side, shouldWarnOnSubmit should be checked together if the any of the form field that contains any code is dirty or updated.